heron/tmaster/src/cpp/manager/tmaster.cpp

```diff
@@ -142,6 +142,14 @@ void TMaster::EstablishTMaster(EventLoop::Status) {
   auto cb = [this](proto::system::StatusCode code) { this->SetTMasterLocationDone(code); };
 
   state_mgr_->SetTMasterLocation(*tmaster_location_, std::move(cb));
+
+  // if zk lost the tmaster location, tmaster quits to bail out and re-establish its location
+  auto cb2 = [this]() {
+    LOG(ERROR) << " lost tmaster location in zk state manager. Bailing out..." << std::endl;
```

Comment:
According to zk semantics, this situation should never happen. If it happens it certainly is a zk client/server bug and we are already in a bad state. better to die and restart fresh